tools/text_normalization/verbalizer.py

```diff
@@ -0,0 +1,195 @@
+    except:
+        pass
+    try:
+        data["year"] = expand_year({"value": data["year"]})
+    except:
+        pass
+    data = {k: data[k] for k in data if k in _date_components_whitelist}
+    result = verbalize(**data)
+    return result.replace("-", " ")
+
+
+def _expand_hundreds(text):
+    number = float(text)
+    if number > 1000 < 10000 and (number % 100 == 0) and (number % 1000 != 0):
+        return _inflect.number_to_words(int(number / 100)) + " hundred"
+    else:
+        return _inflect.number_to_words(text)
+
+
+def _expand_currency(data):
+    currency = _currency_dict[data['currency']]
+    quantity = data['integral'] + ('.' + data['fractional'] if data.get('fractional') else '')
+    magnitude = data.get('magnitude')
+
+    # remove commas from quantity to be able to convert to numerical
+    quantity = quantity.replace(',', '')
+
+    # check for million, billion, etc...
+    if magnitude is not None and magnitude.lower() in _magnitudes:
+        if len(magnitude) == 1:
+            magnitude = _magnitudes_dict[magnitude.lower()]
+        return "{} {} {}".format(_expand_hundreds(quantity), magnitude, currency + 's')
+
+    parts = quantity.split('.')
+    if len(parts) > 2:
+        return quantity + " " + currency + "s"  # Unexpected format
+
+    dollars = int(parts[0]) if parts[0] else 0
+
+    cents = int(parts[1]) if len(parts) > 1 and parts[1] else 0
+    if dollars and cents:
+        dollar_unit = currency if dollars == 1 else currency + 's'
+        cent_unit = 'cent' if cents == 1 else 'cents'
+        return "{} {}, {} {}".format(
+            _expand_hundreds(dollars), dollar_unit, _inflect.number_to_words(cents), cent_unit
+        )
+    elif dollars:
+        dollar_unit = currency if dollars == 1 else currency + 's'
+        return "{} {}".format(_expand_hundreds(dollars), dollar_unit)
+    elif cents:
+        cent_unit = 'cent' if cents == 1 else 'cents'
+        return "{} {}".format(_inflect.number_to_words(cents), cent_unit)
+    else:
+        return 'zero' + ' ' + currency + 's'
+
+
+def expand_money(data):
+    result = _expand_currency(data)
+    return result.replace(',', '').replace('-', ' ').replace(' and ', ' ')
+
+
+def expand_measurement(data):
+    value = float(data["decimal"].replace(",", ""))
+    value_verb = _inflect.number_to_words(data["decimal"]).replace(',', '').replace('-', ' ').replace(' and ', ' ')
+    res = value_verb
+    if data.get("measurement"):
+        measure = _measurements_dict[data["measurement"]]
+        if value <= 1 and measure[-1] == 's':
+            measure = measure[:-1]
+        res += " " + measure
+
+    if data.get("measurement2"):
+        res += " per "
+        measure2 = _measurements_dict[data["measurement2"]]
+        # if measure2[-1] == 's':
+        #     measure2 = measure2[:-1]
+        res += measure2
+    return res
+
+
+def expand_time(data):
+    res = _inflect.number_to_words(data["hour"])
```

Comment:
types + docstring are missing for the file

Reply:
done

Reply:
types are still missing

Reply:
@yzhang123